pkg/virt-operator/resource/apply/apps.go

```diff
@@ -387,3 +398,25 @@ func (r *Reconciler) syncPodDisruptionBudgetForDeployment(deployment *appsv1.Dep
 
 	return nil
 }
+
+func getDesiredApiReplicas(clientset kubecli.KubevirtClient) (replicas int32, err error) {
+	nodeList, err := clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get number of nodes to determine virt-api replicas: %v", err)
+	}
+
+	nodesCount := len(nodeList.Items)
+	// This is a simple heuristic to achieve basic scalability so we could be running on large clusters.
+	// From recent experiments we know that for a 100 nodes cluster, 9 virt-api replicas are enough.
+	// This heuristic is not accurate. It could, and should, be replaced by something more sophisticated and refined
+	// in the future.
+
+	const minReplicas = 2
```

Comment:
Nit: I would move this to the top as a global const for more visibility.

Reply:
As long as this const is only related to this function I don't see a reason to move it to a top level.
Perhaps in the future when we'll scale other components as well?